pkg/daemon/ceph/client/filesystem.go

```diff
@@ -174,6 +177,23 @@ func IsMultiFSEnabled() bool {
 	}
 	return false
 }
+func SetPoolSize(context *clusterd.Context, clusterName, fsName string, dataPool []*model.Pool, metdataPool *model.Pool) error {
+	// updating the dataPool size
+	metaDataPoolname := fsName + metaDataPoolSuffix
+	err := SetPoolProperty(context, clusterName, metaDataPoolname, "size", strconv.Itoa(int(metdataPool.ReplicatedConfig.Size)))
+	if err != nil {
+		return errors.Wrapf(err, "failed to update metadataPool replica for filesystem %s", fsName)
+	}
+	//updating dataPool size
+	dataPoolname := fsName + dataPoolSuffix
+	for i := 0; i < len(dataPool); i++ {
+		err = SetPoolProperty(context, clusterName, dataPoolname+strconv.Itoa(int(i)), "size", strconv.Itoa(int(dataPool[i].ReplicatedConfig.Size)))
```

Comment:
You should move the `if err != nil {` in the loop.

Reply:
My bad, I didn't see that!